src/main/java/appeng/client/gui/widgets/GuiFluidTank.java

```diff
@@ -0,0 +1,106 @@
+
+package appeng.client.gui.widgets;
+
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.GuiButton;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.client.renderer.texture.TextureMap;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.IFluidTank;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+import appeng.api.util.AEColor;
+
+
+@SideOnly( Side.CLIENT )
+public class GuiFluidTank extends GuiButton implements ITooltip
+{
+	private final IFluidTank tank;
+
+	public GuiFluidTank( int buttonId, IFluidTank tank, int x, int y, int w, int h )
+	{
+		super( buttonId, x, y, w, h, "" );
+		this.tank = tank;
+	}
+
+	@Override
+	public void drawButton( final Minecraft mc, final int mouseX, final int mouseY, final float partialTicks )
+	{
+		if( this.visible )
+		{
+			GlStateManager.color( 1.0F, 1.0F, 1.0F, 1.0F );
+			GlStateManager.disableBlend();
+
+			drawRect( this.x, this.y, this.x + this.width, this.y + this.height, AEColor.GRAY.blackVariant | 0xFF000000 );
+
+			if( this.tank != null )
+			{
+				final FluidStack fluid = this.tank.getFluid();
+				if( fluid != null && fluid.amount > 0 )
+				{
+					mc.getTextureManager().bindTexture( TextureMap.LOCATION_BLOCKS_TEXTURE );
+
+					TextureAtlasSprite sprite = mc.getTextureMapBlocks().getAtlasSprite( fluid.getFluid().getStill().toString() );
+					final int scaledHeight = (int) ( this.height * ( (float) fluid.amount / this.tank.getCapacity() ) );
+
+					int iconHeightRemainder = scaledHeight % 16;
+					if( iconHeightRemainder > 0 )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)